targets/unity-v2/source/Authentication/PlayFabAuthService.cs

```diff
@@ -201,28 +203,52 @@ internal void InvokeDisplayAuthentication()
                 OnDisplayAuthentication.Invoke();
         }
 
-        public void Link(AuthTypes linkType)
+        internal void InvokeLoginSuccess(LoginResult loginResult)
         {
-            var auth = AuthStrategies[linkType];
+            if(OnLoginSuccess != null)
+                OnLoginSuccess.Invoke(loginResult);
+        }
+
+        internal void InvokePlayFabError(PlayFabError playFabError)
+        {
+            if(OnPlayFabError != null)
+                OnPlayFabError.Invoke(playFabError);
+            Debug.LogError(playFabError.GenerateErrorReport());
+        }
+
+        public void Link(AuthTypes linkType, AuthKeys authKeys = null)
+        {
+            var auth = _authStrategies[linkType];
+
             if (auth == null)
             {
                 Debug.LogError("Unhandled link type: " + linkType);
                 return;
             }
-            
-            auth.Link(this);
+
+            auth.Link(this, authKeys);
         }
 
-        public void Unlink(AuthTypes unlinkType)
+        public void Unlink(AuthTypes unlinkType, AuthKeys authKeys = null)
         {
-            var auth = AuthStrategies[unlinkType];
+            var auth = _authStrategies[unlinkType];
+
             if (auth == null)
             {
                 Debug.LogError("Unhandled unlink type: " + unlinkType);
                 return;
             }
-            
-            auth.Unlink(this);
+
+            auth.Unlink(this, authKeys);
         }
     }
-}
\ No newline at end of file
+
+    [Serializable]
+    public sealed class AuthKeys
+    {
+        public string AuthTicket;
```

Comment:
AuthTicket is inherently bound to the ```AuthTypes authType``` that almost always comes with this.
I really like this structure, and I think moving the authType adjacent to AuthTicket would make it perfect.